src/finch/tensor.py

```diff
@@ -154,14 +157,27 @@ def __mod__(self, other):
     def __pow__(self, other):
         return self._elemwise_op("^", other)
 
-    def __matmul__(self, other):
-        # TODO: Implement and use mul instead of tensordot
-        # https://github.com/FinchTensor/finch-tensor-python/pull/22#issuecomment-2007884763
-        if self.ndim != 2 or other.ndim != 2:
-            raise ValueError(
-                f"Both tensors must be 2-dimensional, but are: {self.ndim=} and {other.ndim=}."
+    @compiled()
+    def __matmul__(self, other: "Tensor") -> "Tensor":
+        if self.ndim < 2 or other.ndim < 2:
+            raise NotImplementedError(
+                "`@` is not implemented for arrays with less than two dimensions."
             )
-        return tensordot(self, other, axes=((-1,), (-2,)))
```

Comment:
As I checked today, with Finch 1.0 we don't need this hack - `tensordot` is fine to use.

Reply:
This is more general as well; supports tensors with ndim >= 2.

Reply:
Do we know if tensordot supports ndim > 2? If not I’d prefer to keep it this way.

Reply:
tensordot supports ndim > 2

Reply:
Done in https://github.com/finch-tensor/finch-tensor-python/pull/83/commits/1f6513c7da4b1516c34e75484e912338eaa7c7b6.